### ISSUE TITLE
NetBeans on OS X does not stop Boot app that it's launched with exec-maven-plugin:exec

### ISSUE THREAD

**Issue by twothe** [spring-projects/spring-boot#4457]
When doing my first steps with spring-boot I learned that when using the default Maven target to run the application, the Tomcat instance is not properly shutdown. The exact same project and code works perfectly fine on a Windows machine, so this has to be a Mac issue. As project I used the (unmodified) spring-boot Greeting REST-Service-example.

Running the Maven default target ``process-classes org.codehaus.mojo:exec-maven-plugin:exec`` has the issue, running the target ``spring-boot:run`` does not have this issue.
The later target cannot be automatically configured by some IDEs as it is a non-standard Maven target and therefore not a solution.

**Comment by philwebb** [comment]
What do you mean by "the Tomcat instance is not properly shutdown"? Can you be a little more specific about what you expect to happen and the exact steps that you are taking to run and terminate the app.

**Comment by twothe** [comment]
The app is started by running Maven with the target as above. The application is terminated with a standard kill command.

Expected: when terminating the Maven run process, Tomcat shuts down and terminates as well.
What actually happens: the run process terminates, but the Tomcat instance continues to linger in the background and never terminates until terminated manually via kill.

**Comment by wilkinsona** [comment]
@twothe As @philwebb requested, I think we still need to _exact_ steps that you're taking. If I try to run the [REST Service guide](https://spring.io/guides/gs/rest-service/) unmodified with `process-classes org.codehaus.mojo:exec-maven-plugin:exec` it fails:

```
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.3.2:exec (default-cli) on project gs-rest-service: The parameters 'executable' for goal org.codehaus.mojo:exec-maven-plugin:1.3.2:exec are missing or invalid -> [Help 1]
```

What other changes have you made? A sample project with precise steps to reproduce the problem would be a big help here.

**Comment by twothe** [comment]
1. Download Netbeans from Netbeans.org (Java version is sufficient)
2. Checkout or copy the Spring-boot Greeting example.
3. Hit run, the appropriate Maven task will be executed.
4. Click the stop button. The run process terminates, the Tomcat still lingers.
5. Check Tomcat PID with lsof in terminal and kill it.

**Comment by wilkinsona** [comment]
The problem is that two JVMs are launched: one that runs Maven and one that runs the Boot app. NetBeans managed to shutdown the former but fails to shutdown the latter. Apparently [this is a known issue](https://netbeans.org/bugzilla/show_bug.cgi?id=135475). While the state of the issue is `RESOLVED FIXED`, that doesn't appear to be the case. See the comments toward the end:

> I still see this with 7.3 and 7.4. Running projects and then stopping them leaves the Java process in the background. NB thinks it's killed.

>  This is definitely still occurring with Netbeans 7.4. 

> We can only reliably kill the maven JVM itself.

[This issue] and the issue it duplicates may also be of interest. It suggests the problem should be fixed in Netbeans 8.1 but that doesn't appear to be the case.

In short, this problem is a limitation of Netbeans. Perhaps there's a way to convince Netbeans to run your application's main method directly rather than via Maven's exec plugin?
